cpp/src/parquet/encoding.cc

```diff
@@ -127,6 +129,21 @@ class PlainEncoder : public EncoderImpl, virtual public TypedEncoder<DType> {
   }
 
  protected:
+  template <typename ArrayType>
+  void PutBinaryArray(const ArrayType& array) {
+    const int64_t total_bytes =
+        array.value_offset(array.length()) - array.value_offset(0);
+    PARQUET_THROW_NOT_OK(sink_.Reserve(total_bytes + array.length() * sizeof(uint32_t)));
```

Comment:
is a check for int32 overflow needed here here?

Reply:
I don't think so (these are `int64_t` calculations AFAICT). However, I should fix the XXX below :-)